gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapperTest.java

```diff
@@ -230,7 +227,82 @@ public void When_MappingMedicationRequestWithAvailabilityTimeOnlyInEhrExtract_Ex
                 .findFirst()
                 .get();
 
-        assertThat(medicationRequest.getAuthoredOnElement().getValue()).isEqualTo(expectedAvailabilityTime.getValue());
+        assertThat(medicationRequest.getAuthoredOnElement().getValue()).isNull();
+    }
+
+    @Test
+    public void When_MappingMedicationRequestWithAuthoredOnValidDate_Expect_AuthoredOnToUseMedicationAvailabilityTime() {
+        var ehrExtract = unmarshallEhrExtract("ehrExtract_hasAuthorTime.xml");
+        var expectedAuthoredOn = DateFormatUtil.parseToDateTimeType("20100115");
+
+        when(medicationRequestPlanMapper.mapToPlanMedicationRequest(any(), any(), any(), any())).thenReturn(new MedicationRequest());
+        when(medicationRequestOrderMapper.mapToOrderMedicationRequest(any(), any(), any(), any())).thenReturn(new MedicationRequest());
+        when(medicationStatementMapper.mapToMedicationStatement(any(), any(), any(), any(), any())).thenReturn(new MedicationStatement());
+        when(medicationMapper.createMedication(any())).thenReturn(new Medication());
+
+        var resources = medicationRequestMapper.mapResources(ehrExtract, (Patient) new Patient().setId(PATIENT_ID), List.of(),
+                PRACTISE_CODE);
+
+        assertThat(resources.size()).isEqualTo(EXPECTED_RESOURCES_MAPPED);
+
+        var medicationRequest = resources
+                .stream()
+                .filter(resource -> ResourceType.MedicationRequest.equals(resource.getResourceType()))
+                .map(MedicationRequest.class::cast)
+                .findFirst()
+                .get();
+
+        assertThat(medicationRequest.getAuthoredOnElement().getValue()).isEqualTo(expectedAuthoredOn.getValue());
+    }
+
+    @Test
+    public void When_MappingMedicationRequestWithAuthoredOnValidDateInExtractAndComposition_Expect_AuthoredOnToUseCompositionValue() {
```

Comment:
Is this test also just using the value from the `MedicationStatement` too or am I being thick?

Reply:
Title was outdated as well.